### PR TITLE
Moving Pylint guidelines checker from sdk-for-python to here

### PR DESCRIPTION
**Pull request by scbedd**
@tjprescott just FYI

Can you please try using the external package (instead of the inserted into path as seen in `rcfile` usage?)

- [x]  There are a couple test failures that are outstanding that I need to address (they are pointed at non-existent guidelines pages now due to rework of azure-sdk github.io page)
- [x] I also need to rename from `pylint_custom_plugin` to `pylint_guidelines_checker`.

### CODE REVIEW

File: tools/pylint-extensions/pylint_custom_plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+## Linting the Guidelines
+
+In order to lint for the guidelines, you must make sure you are using the pylintrc file.
+It is recommended you run pylint at the library package level to be consistent with how the CI runs pylint.
+
+Check that you are running pylint version >=2.5.2 and astroid version >=2.4.1.
+
+**How to run pylint locally using the pylintrc:**
+
+1. Run pylint at the root of the repo and it will automatically find the pylintrc:
```
Comment (scbedd) [inline tools/pylint-extensions/pylint_custom_plugin/README.md:10]: All of this needs to be updated. I'll do so before this merges.

**Comment by tjprescott** [comment]
Adding @l0lawrence as reviewer since she's working on pylint plugin stuff as well.

**Comment by tjprescott** [comment]
Also I think @l0lawrence already had a PR that fixed the broken links.

File: tools/pylint-extensions/pylint_custom_plugin/ci.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  parameters:
+    PythonVersion: '3.10.2'
+    PackagePath: 'tools/pylint-extensions/pylint_custom_plugin'
+    FeedName: 'public/azure-sdk-for-python'
+    ArtifactName: 'azsdkpylintplugin'
+    PackageName: 'Azure SDK Pylint Plugin'
+    TestSteps:
+    - template: /eng/pipelines/templates/steps/use-python-version.yml
+      parameters:
+        versionSpec: '3.10.2'
```
Comment (weshaggard) [inline tools/pylint-extensions/pylint_custom_plugin/ci.yml:35]: PythonVersion

Comment (scbedd) [reply tools/pylint-extensions/pylint_custom_plugin/ci.yml:35]: What? [no it definitely takes versionSpec](https://github.com/Azure/azure-sdk-tools/blob/2218892688c1c2cd46e118b16b951233b667d76a/eng/pipelines/templates/steps/use-python-version.yml#L2)

Or are you saying we should make this a variable, I'll do that 👍

Comment (weshaggard) [reply tools/pylint-extensions/pylint_custom_plugin/ci.yml:35]: I'm just calling out we already have a variable with the same value above that we should use here.

Comment (scbedd) [reply tools/pylint-extensions/pylint_custom_plugin/ci.yml:35]: The code comment no longer highlights exactly which line you're talking about, so it's super confusing :P

Comment (weshaggard) [reply tools/pylint-extensions/pylint_custom_plugin/ci.yml]: You got it.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**